app/controllers/admin/prosecution_cases_controller.rb

```diff
@@ -53,6 +59,13 @@ def result
       end
     end
 
+    def associate_court_application
+      p_case = ProsecutionCase.find(params[:id])
+      court_application = CourtApplication.find(params[:court_application_id])
+      court_application.prosecution_cases << p_case
+      redirect_to admin_court_application_path(court_application), flash: { notice: "Case associated" }
```

Comment:
I think the redirection should be removed from this method whose purpose is to add the prosecution case to the court_application. 
Actually this method could be moved into the `ProsecutionCase`

Reply:
I've separated out the model logic from the rest of the controller action. I didn't want it polluting the global ProsecutionCase model with situation-specific business logic so I've put it into a private method in the controller.